src/libraries/System.Net.Http/src/System/Net/Http/ByteArrayHelpers.cs

```diff
@@ -22,16 +25,33 @@ internal static bool EqualsOrdinalAsciiIgnoreCase(string left, ReadOnlySpan<byte
                 uint charA = left[i];
                 uint charB = right[i];
 
-                unchecked
+                // We're only interested in ASCII characters here.
+                if ((charA - 'a') <= ('z' - 'a'))
+                    charA -= ('a' - 'A');
+                if ((charB - 'a') <= ('z' - 'a'))
+                    charB -= ('a' - 'A');
+
+                if (charA != charB)
```

Comment:
```suggestion
                return charA == charB;
```

(Although #32632 might help with this in the end.)

Reply:
That would change the meaning and break the function; it shouldn't return here if they're equal, only if they're not equal.  If they're equal it should keep iterating.

Reply:
D'oh. I really need more coffee before I start making code reviews.

Reply:
:smile: